interfaces/acados_template/acados_template/zoro_description.py

```diff
@@ -70,6 +70,21 @@ class ZoroDescription:
     idx_uh_t: list = field(default_factory=list)
     idx_lh_e_t: list = field(default_factory=list)
     idx_uh_e_t: list = field(default_factory=list)
+    # Inputs:
+    input_P0_diag: bool = False
+    """Determines if diag(P0) is an input to the custom update function"""
+    input_P0: bool = True
+    """Determines if diag(P0) is an input to the custom update function"""
```

Comment:
Personally, I think this should be designed such that the user cannot set an impossible state (i.e. both as true). This could be done by only providing one attribute where the user specifies the input P0 matrix traits using a string or an enum.
```suggestion
    input_P0: bool = None
    """Determines if P0 is an input to the function. Allowed options are [None | "diag" | "full_matrix"]"""
```

and in the custom update
```cpp
{%- if zoro_description.input_P0 == "diag" %}
     offset += nx;  // P0_diag
{%- elif zoro_description.input_P0 == "full_matrix" %}
     ...
```
Or is this not doable with the code generator?

Reply:
Hi @naefjo,
thanks for the quick feedback!
I thought of this combination by adding a sanity check here
https://github.com/acados/acados/pull/1054/files#diff-d6902a8bd481c9f7d9cabcb0656e1fdfe1ba9ab2d764bacf19f065dd6aba5333R108-R109
I think it is nice to have the `input_*` fields to be all of the same type and thus would stick to the Boolean, especially because it is already implemented like this.

Reply:
Alright, makes sense :)